### PR TITLE
Updated php and symfony versions

### PR DESCRIPTION
**Pull request by Onkel-Martin**
https://jira.itkdev.dk/browse/MUHU-190

upgraded to php 8.1 and symfony 6.2

### CODE REVIEW

**Review by rimi-itk — COMMENTED** [review]
`.DS_Store` files should not be committed. Add them to your global git ignore file (cf. https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/).

File: README.md [hunk 1]
```diff
@@ -12,7 +11,7 @@ Profile = Internal events:
 
 Profile = External and free events:
   * 'EventsExtern': is defined as events with productionOnline from Yesplan = 1
-
+
```
Comment (rimi-itk) [inline README.md]: Configure your editor to delete trailing whitespace from lines when saving a file.

File: composer.json [hunk 2]
```diff
@@ -1,30 +1,34 @@
...
-        "symfony/orm-pack": "^1.0",
-        "symfony/yaml": "^5.0"
-    },
-    "require-dev": {
-        "friendsofphp/php-cs-fixer": "^2.16",
-        "symfony/maker-bundle": "^1.15"
+        "doctrine/doctrine-bundle": "^2.8",
+        "doctrine/doctrine-migrations-bundle": "^3.2",
+        "doctrine/orm": "^2.14",
+        "friendsofphp/php-cs-fixer": "^3.14",
```
Comment (rimi-itk) [inline composer.json]: This should be moved to `require-dev`.

File: src/Controller/MailerController.php [hunk 3]
```diff
@@ -24,36 +16,35 @@ public function __construct(array $mailerOptions, MailerInterface $mailer)
...
+      */
+     public function sendEmail(string $subject, string $message): void
+     {
+         $email = (new Email())
+         ->from($this->options['mail_from'])
+         ->to($this->options['mail_to'])
+         // ->cc('cc@example.com')
+         // ->bcc('bcc@example.com')
+         // ->replyTo('fabien@example.com')
+         // ->priority(Email::PRIORITY_HIGH)
```
Comment (rimi-itk) [inline src/Controller/MailerController.php]: Remove code that is not used.

File: src/Entity/YesplanEvent.php [hunk 4]
```diff
@@ -10,150 +10,98 @@
...
+#[ORM\Entity(repositoryClass: YesplanEventRepository::class)]
 class YesplanEvent
 {
     /*
-     * Hook timestampable behavior
-     * updates createdAt, updatedAt fields
-     */
+       * Hook timestampable behavior
+       * updates createdAt, updatedAt fields
+       */
```
Comment (rimi-itk) [inline src/Entity/YesplanEvent.php]: Not indented correctly.

File: src/Repository/YesplanEventRepository.php [hunk 5]
```diff
@@ -52,7 +52,7 @@ public function configureOptions(OptionsResolver $resolver): void
     public function findOldEvents(): array
     {
         $entityManager = $this->getEntityManager();
-        $timeNow = new DateTime('NOW');
+        $timeNow = new \DateTime('NOW');
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Make a habit of using `DateTimeImmutable`. Using `DateTime` (which is mutable) sometimes makes things harder to do (not in this case however).

File: src/Repository/YesplanEventRepository.php [hunk 6]
```diff
@@ -70,18 +70,20 @@ public function findOldEvents(): array
...
-            ';
-        $stmt = $conn->prepare($sql);
-        $stmt->execute(['profileIdIntern' => $this->options['yesplan_intern_profile_id']]);
-
-        return $stmt->fetchAll();
+        $query = $entityManager->createQuery(
+            'SELECT y.id
+            FROM App\Entity\YesplanEvent y
+            LEFT JOIN App\Entity\AsanaEvent a WITH y.id=a.id
+            WHERE y.productionOnline = true AND  (a.createdInNewEvents is null OR a.createdInNewEvents = 0)
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: `createdInNewEvents` is a boolean value and should not be compared to an integer. Use `… OR a.createdInNewEvents = false`.

File: src/Repository/YesplanEventRepository.php [hunk 7]
```diff
@@ -70,18 +70,20 @@ public function findOldEvents(): array
      */
     public function findNewProductionOnlineEvents(): array
     {
-        //get all events with productiononline = 1 not already created in Asana
+        // get all events with productiononline = 1 not already created in Asana
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: This comment (on the return value) does not match the one in the PHPDoc block for the function. Update the PHPDoc block and clean up this comment.

File: src/Repository/YesplanEventRepository.php [hunk 8]
```diff
@@ -70,18 +70,20 @@ public function findOldEvents(): array
...
-        return $stmt->fetchAll();
+        $query = $entityManager->createQuery(
+            'SELECT y.id
+            FROM App\Entity\YesplanEvent y
+            LEFT JOIN App\Entity\AsanaEvent a WITH y.id=a.id
+            WHERE y.productionOnline = true AND  (a.createdInNewEvents is null OR a.createdInNewEvents = 0)
+            AND y.profileId = :profileIdIntern
+            '
+        )->setParameter('profileIdIntern', $this->options['yesplan_intern_profile_id']);
+        // returns an array of event id's
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: State this in the PHPDoc block for the function.

File: src/Repository/YesplanEventRepository.php [hunk 9]
```diff
@@ -91,18 +93,23 @@ public function findNewProductionOnlineEvents(): array
      */
     public function findNewProductionOnlineIncludingGratisandExternEvents(): array
     {
-        //get all events with productiononline = 1 not already created in Asana
+        // get all events with productiononline = 1 not already created in Asana
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: This comment (on the return value) does not match the one in the PHPDoc block for the function. Update the PHPDoc block and clean up this comment.

File: src/Repository/YesplanEventRepository.php [hunk 10]
```diff
@@ -91,18 +93,23 @@ public function findNewProductionOnlineEvents(): array
...
+        $entityManager = $this->getEntityManager();
 
-        return $stmt->fetchAll();
+        $query = $entityManager->createQuery(
+            'SELECT y.id
+               FROM App\Entity\YesplanEvent y
+               LEFT JOIN App\Entity\AsanaEvent a WITH y.id=a.id
+               WHERE y.productionOnline = true
+                AND  (a.createdInNewEvents is null OR a.createdInNewEvents = 0)
+                AND (a.createdInNewEvents is null OR a.createdInNewEvents = 0)
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: `createdInNewEvents` is a boolean. Compare with a boolean.

File: src/Repository/YesplanEventRepository.php [hunk 11]
```diff
@@ -91,18 +93,23 @@ public function findNewProductionOnlineEvents(): array
...
+                AND (y.profileId = :profileIdEkstern OR y.profileId = :profileIdGratis)
+              '
+        )->setParameters(['profileIdEkstern' => $this->options['yesplan_external_profile_id'], 'profileIdGratis' => $this->options['yesplan_free_profile_id']]);
+        // returns an array of event id's
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Remove this line and fix the PHPDoc block for the function.

File: src/Repository/YesplanEventRepository.php [hunk 12]
```diff
@@ -112,17 +119,21 @@ public function findNewProductionOnlineIncludingGratisandExternEvents(): array
...
+            'SELECT y.id
+            FROM App\Entity\YesplanEvent y
+            LEFT JOIN App\Entity\AsanaEvent a
+            WITH y.id=a.id
+            WHERE y.eventOnline = true
+            AND (a.createdInNewEventsOnline is null OR a.createdInNewEventsOnline = 0)
+            AND y.profileId = :profileId
+            '
+        )->setParameter('profileId', $this->options['yesplan_intern_profile_id']);
+        // returns an array of event id's
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Remove this line and fix the PHPDoc block for the function.

File: src/Repository/YesplanEventRepository.php [hunk 13]
```diff
@@ -133,65 +144,72 @@ public function findNewEventOnlineEvents(): array
...
+            'SELECT y.id
+            FROM App\Entity\YesplanEvent y
+            LEFT JOIN App\Entity\AsanaEvent a
+            WITH y.id=a.id
+            WHERE y.capacityPercent >= 90
+            AND  (a.createdInFewTickets is null OR a.createdInFewTickets = 0)
+            AND y.profileId = :profileId
+            '
+        )->setParameter('profileId', $this->options['yesplan_intern_profile_id']);
+        // returns an array of event id's
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Remove this line and fix the PHPDoc block for the function.

File: src/Repository/YesplanEventRepository.php [hunk 14]
```diff
@@ -133,65 +144,72 @@ public function findNewEventOnlineEvents(): array
...
-
-        $conn = $this->getEntityManager()->getConnection();
-
-        $sql = '
-        SELECT y.id FROM yesplan_event y LEFT JOIN asana_event a ON y.id=a.id WHERE y.capacity_percent <= 75 AND  (a.created_in_last_minute is null OR a.created_in_last_minute = 0) AND y.event_date <= :nowPlus3Weeks AND y.profile_id = :profileId
-            ';
-        $stmt = $conn->prepare($sql);
-        $stmt->execute(['nowPlus3Weeks' => date_format($nowPlus3Weeks, 'Y-m-d H:i:s'), 'profileId' => $this->options['yesplan_intern_profile_id']]);
+        $nowPlus3Weeks = new \DateTime('NOW');
+        $nowPlus3Weeks->add(new \DateInterval('P21D'));
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Rewrite as

```php
$nowPlus3Weeks = new DateTimeImmutable('+3 weeks'));
````

File: src/Repository/YesplanEventRepository.php [hunk 15]
```diff
@@ -133,65 +144,72 @@ public function findNewEventOnlineEvents(): array
...
+                LEFT JOIN App\Entity\AsanaEvent a
+                WITH y.id=a.id
+                WHERE y.capacityPercent <= 75
+                AND  (a.createdInLastMinute is null OR a.createdInLastMinute = 0)
+                AND y.eventDate <= :nowPlus3Weeks
+                AND y.profileId = :profileId
+
+            '
+        )->setParameters(['nowPlus3Weeks' => date_format($nowPlus3Weeks, 'Y-m-d H:i:s'), 'profileId' => $this->options['yesplan_intern_profile_id']]);
+        // returns an array of event id's
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Remove this line and fix the PHPDoc block for the function.

File: src/Repository/YesplanEventRepository.php [hunk 16]
```diff
@@ -133,65 +144,72 @@ public function findNewEventOnlineEvents(): array
...
+                LEFT JOIN App\Entity\AsanaEvent a
+                WITH y.id=a.id
+                WHERE ((a.createdInCalendar is null OR a.createdInCalendar = 0)
+                  OR y.inSaleDateUpdated = 1
+                  OR y.inPresaleDateUpdated = 1
+                  OR y.eventDateUpdated = 1)
+             AND y.profileId = :profileId
+            '
+        )->setParameter('profileId', $this->options['yesplan_intern_profile_id']);
+        // returns an array of event id's
```
Comment (rimi-itk) [inline src/Repository/YesplanEventRepository.php]: Remove this line and fix the PHPDoc block for the function.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Some minor changes needed.

File: .gitignore [hunk 17]
```diff
@@ -1,4 +1,5 @@
-
+.DS_Store
```
Comment (rimi-itk) [inline .gitignore:1]: You must also remove any committed `.DS_Store` files.

File: composer.json [hunk 18]
```diff
@@ -85,6 +81,10 @@
         }
     },
     "require-dev": {
-        "symfony/maker-bundle": "^1.48"
+        "symfony/maker-bundle": "^1.48",
+        "doctrine/doctrine-bundle": "^2.8",
+        "doctrine/doctrine-migrations-bundle": "^3.2",
+        "doctrine/orm": "^2.14",
```
Comment (rimi-itk) [inline composer.json]: The doctrine packages should not be under `require-dev`. They are not development dependencies, but required in production as well.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

**Review by rimi-itk — APPROVED** [review]

File: .gitignore [hunk 19]
```diff
@@ -13,4 +13,5 @@
 ###> friendsofphp/php-cs-fixer ###
 /.php_cs
 /.php_cs.cache
+/.php-cs-fixer.cache
```
Comment (rimi-itk) [inline .gitignore]: I have a feeling that one of `/.php_cs.cache` and `/.php-cs-fixer.cache` can be removed.

File: docker-compose.yml [hunk 20]
```diff
@@ -20,58 +21,54 @@ services:
...
     networks:
       - app
+      - frontend
     ports:
       - "1025"
       - "8025"
+    labels:
+      - "traefik.enable=true"
+      - "traefik.docker.network=frontend"
+      - "traefik.http.routers.${COMPOSE_PROJECT_NAME}-http.rule=Host(`${COMPOSE_DOMAIN}`)"
```
Comment (rimi-itk) [inline docker-compose.yml]: Add newline at end of file (if it's missing in the source template we should clean that up).

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 20 above)
Comment (jekuaitk) [reply docker-compose.yml]: Sure, added newline.

File: .gitignore (hunk 19 above)
Comment (jekuaitk) [reply .gitignore]: `/.php_cs.cache` can indeed be removed and has been removed.

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**